base/linalg/symmetric.jl

```diff
@@ -218,6 +248,10 @@ A_mul_B!{T<:BlasComplex,S<:StridedMatrix}(C::StridedMatrix{T}, A::StridedMatrix{
 *(A::HermOrSym, B::HermOrSym) = full(A)*full(B)
 *(A::StridedMatrix, B::HermOrSym) = A*full(B)
 
+for T in (:Symmetric, :Hermitian), op in (:+, :-, :*, :/)
+    @eval ($op)(A::$T, x::Number) = ($T)(($op)(A.data, x), Symbol(A.uplo))
```

Comment:
I had thought about that, but there are cases where operating with a complex value could still produce a real diagonal. But restricting to `Real` does seem the safer option.